src/backend/access/external/external.c

```diff
@@ -661,14 +671,20 @@ create_external_scan_uri_list(ExtTableEntry *ext, bool *ismasteronly)
 							   uri->protocol == URI_GPFDISTS ||
 							   uri->protocol == URI_CUSTOM))
 	{
-		if ((strcmp(on_clause, "COORDINATOR_ONLY") == 0) && (uri->protocol == URI_CUSTOM))
+		if ((strcmp(on_clause, "COORDINATOR_ONLY") == 0 || IS_UTILITY_OR_SINGLENODE(Gp_role)) && (uri->protocol == URI_CUSTOM))
 		{
 			const char *uri_str = strVal(linitial(ext->urilocations));
 			segdb_file_map[0] = pstrdup(uri_str);
 			*ismasteronly = true;
 		}
 		else
 		{
+			/* We currently don't support gpfdist in SINGLE NODE mode. */
+			if (IS_UTILITY_OR_SINGLENODE(Gp_role))
```

Comment:
what about utility mode ?

Reply:
gpfdist cannot work correctly in utility mode now. It will return nothing even data exists.